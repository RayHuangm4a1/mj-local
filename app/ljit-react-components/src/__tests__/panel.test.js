import React from 'react';
import { shallow, mount, } from 'enzyme';
import Panel from '../components/panel';

describe('Panel', () => {
	it('should renders correctly', () => {
		const className = 'mock-class';
		const headerTitle = 'mock-Headertitle';
		const headerRight = 'mock-headerRight';
		const content = 'mock-content';
		const footer = 'mock-footer';

		const wrapper = shallow(
			<Panel
				className={className}
				headerTitle={headerTitle}
				headerRight={headerRight}
				content={content}
				footer={footer}
			/>
		);

		expect(wrapper).toMatchSnapshot();
	});

	it('should mount in a full DOM', () => {
		const className = 'mock-class';
		const headerTitle = 'mock-Headertitle';
		const headerRight = 'mock-headerRight';
		const content = 'mock-content';
		const footer = 'mock-footer';

		const wrapper = mount(
			<Panel
				className={className}
				headerTitle={headerTitle}
				headerRight={headerRight}
				content={content}
				footer={footer}
			/>
		);

		expect(wrapper.props().className).toBe(className);
		expect(wrapper.props().headerTitle).toBe(headerTitle);
		expect(wrapper.props().headerRight).toBe(headerRight);
		expect(wrapper.props().content).toEqual(content);
		expect(wrapper.props().footer).toEqual(footer);
	});

	it('should be selectable by class ljit-list', () => {
		const wrapper = shallow(<Panel />);

		expect(wrapper.hasClass('ljit-panel')).toEqual(true);
	});

	it('should be selectable by custom class', () => {
		const className = 'mock-class';
		const wrapper = shallow(<Panel className={className} />);

		expect(wrapper.hasClass(className)).toEqual(true);
	});

	it('should renders react node content', () => {
		const content = (
			<div>
				<div>mock text 1</div>
				<div>mock text 2</div>
				<div>mock text 3</div>
			</div>
		);
		const wrapper = shallow(<Panel content={content} />);

		expect(wrapper.contains(content)).toEqual(true);
	});

	it('should renders react node footer', () => {
		const footer = (
			<div>
				<div>mock text 1</div>
				<div>mock text 2</div>
				<div>mock text 3</div>
			</div>
		);
		const wrapper = shallow(<Panel footer={footer} />);

		expect(wrapper.contains(footer)).toEqual(true);
	});
});
