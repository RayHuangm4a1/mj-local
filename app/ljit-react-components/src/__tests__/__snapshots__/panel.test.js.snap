// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Panel should renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Panel
    className="mock-class"
    content="mock-content"
    footer="mock-footer"
    headerRight="mock-headerRight"
    headerTitle="mock-Headertitle"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="ljit-panel__header"
        >
          <span
            className="ljit-panel__title"
          >
            mock-Headertitle
          </span>
          <div
            className="ljit-panel__right"
          >
            mock-headerRight
          </div>
        </div>,
        <div
          className="ljit-panel__content"
        >
          mock-content
        </div>,
        <div
          className="ljit-panel__footer"
        >
          mock-footer
        </div>,
      ],
      "className": "ljit-panel mock-class",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <span
              className="ljit-panel__title"
            >
              mock-Headertitle
            </span>,
            <div
              className="ljit-panel__right"
            >
              mock-headerRight
            </div>,
          ],
          "className": "ljit-panel__header",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mock-Headertitle",
              "className": "ljit-panel__title",
            },
            "ref": null,
            "rendered": "mock-Headertitle",
            "type": "span",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mock-headerRight",
              "className": "ljit-panel__right",
            },
            "ref": null,
            "rendered": "mock-headerRight",
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mock-content",
          "className": "ljit-panel__content",
        },
        "ref": null,
        "rendered": "mock-content",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "mock-footer",
          "className": "ljit-panel__footer",
        },
        "ref": null,
        "rendered": "mock-footer",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="ljit-panel__header"
          >
            <span
              className="ljit-panel__title"
            >
              mock-Headertitle
            </span>
            <div
              className="ljit-panel__right"
            >
              mock-headerRight
            </div>
          </div>,
          <div
            className="ljit-panel__content"
          >
            mock-content
          </div>,
          <div
            className="ljit-panel__footer"
          >
            mock-footer
          </div>,
        ],
        "className": "ljit-panel mock-class",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <span
                className="ljit-panel__title"
              >
                mock-Headertitle
              </span>,
              <div
                className="ljit-panel__right"
              >
                mock-headerRight
              </div>,
            ],
            "className": "ljit-panel__header",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mock-Headertitle",
                "className": "ljit-panel__title",
              },
              "ref": null,
              "rendered": "mock-Headertitle",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mock-headerRight",
                "className": "ljit-panel__right",
              },
              "ref": null,
              "rendered": "mock-headerRight",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mock-content",
            "className": "ljit-panel__content",
          },
          "ref": null,
          "rendered": "mock-content",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "mock-footer",
            "className": "ljit-panel__footer",
          },
          "ref": null,
          "rendered": "mock-footer",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
